nixos/modules/services/x11/window-managers/xmonad.nix

```diff
@@ -108,7 +120,7 @@ in {
               { modMask = mod4Mask -- Use Super instead of Alt
               , terminal = "urxvt" }
               `additionalKeys`
-              [ ( (mod4Mask,xK_r), compileRestart )
+              [ ( (mod4Mask,xK_r), writeStateToFile >> compileRestart )
```

Comment:
good catch adding `writeStateToFile`, I forgot about layout in my original pr..
But we probably want to add that in `compileRestart` itself instead of here, I think.

Reply:
I created bindings for `compileRestart` both with and without `writeStateToFile` since I occasionally (though very rarely) want to clear out the state (when I've made a mess of things while experimenting with my xmonad config).

Maybe it should accept a Bool parameter like `restart` does.

Reply:
I was thinking of something like:
```haskell
compileRestart :: X ()
compileRestart = whenX (recompile True)
  $ writeStateToFile *> catchIO ( do
     dir  <- getXMonadDataDir
     args <- getArgs
     executeFile (dir </> compiledConfig) False args Nothing )
```

Reply:
In my opinion though, it's cleaner and clearer if `compileRestart` remains as it was and the call site handles writing (or skipping) the state file.

Reply:
@xaverdh I took your suggestion and added a Bool `resume` parameter similar to the one used by `restart`. 88eee47

Reply:
The commit you pushed is the cleanest solution I think, because its the same signature as `restart "xmonad"` :)